contrib/endpoints/src/api_manager/api_manager_impl.cc

```diff
@@ -188,12 +213,19 @@ utils::Status ApiManagerImpl::GetStatistics(
   return utils::Status::OK;
 }
 
+void ApiManagerImpl::AddPendingCheckReportCallback(
+    std::function<void(utils::Status)> callback) {
+  pending_check_report_callbacks_.push_back(callback);
+}
+
 std::unique_ptr<RequestHandlerInterface> ApiManagerImpl::CreateRequestHandler(
     std::unique_ptr<Request> request_data) {
-  std::string config_id = service_selector_->Select();
   return std::unique_ptr<RequestHandlerInterface>(
-      new RequestHandler(check_workflow_, service_context_map_[config_id],
-                         std::move(request_data)));
+      (config_loading_status_.code() == Code::UNAVAILABLE)
+          ? new RequestHandler(this, check_workflow_, std::move(request_data))
+          : new RequestHandler(check_workflow_,
+                               GetServiceContext(SelectConfigId()),
```

Comment:
not doing check here,  always construct it as the first form.  doing the status checking at Check() call.
My concern is race condition:  when checking here,  ExecutePending() is not called,   at Check(),  it is called.